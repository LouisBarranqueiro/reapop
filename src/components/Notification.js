import React, {Component} from 'react'
import PropTypes from 'prop-types'
import {connect} from 'react-redux'
import {Timer, mapObjectValues} from '../helpers'
import {removeNotification} from '../store/notifications'
import {POSITIONS} from '../constants'

/**
 * Create a timer
 * @param {Number} dismissAfter
 * @param {Function} callback
 * @returns {Function|null} a Timer
 */
function createTimer(dismissAfter, callback) {
  if (dismissAfter > 0) {
    return new Timer(dismissAfter, callback)
  }
  return null
}

export class Notification extends Component {
  static propTypes = {
    className: PropTypes.shape({
      main: PropTypes.string.isRequired,
      wrapper: PropTypes.string.isRequired,
      meta: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      message: PropTypes.string.isRequired,
      imageContainer: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
      icon: PropTypes.string.isRequired,
      status: PropTypes.func.isRequired,
      dismissible: PropTypes.string.isRequired,
      closeButtonContainer: PropTypes.string.isRequired,
      closeButton: PropTypes.string.isRequired,
      buttons: PropTypes.func.isRequired,
      button: PropTypes.string.isRequired,
      buttonText: PropTypes.string.isRequired
    }).isRequired,
    notification: PropTypes.shape({
      id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
      title: PropTypes.string,
      message: PropTypes.string,
      image: PropTypes.string,
      status: PropTypes.string.isRequired,
      position: PropTypes.oneOf(mapObjectValues(POSITIONS)),
      dismissAfter: PropTypes.number.isRequired,
      dismissible: PropTypes.bool.isRequired,
      onAdd: PropTypes.func,
      onRemove: PropTypes.func,
      closeButton: PropTypes.bool.isRequired,
      buttons: PropTypes.arrayOf(
        PropTypes.shape({
          name: PropTypes.string.isRequired,
          onClick: PropTypes.func
        })
      ).isRequired,
      allowHTML: PropTypes.bool.isRequired
    }).isRequired,
    removeNotification: PropTypes.func.isRequired
  };

  /**
   * Init timer
   * @param {Object} props
   * @returns {void}
   */
  constructor(props) {
    const {dismissAfter} = props.notification
    super(props)
    this.state = {
      timer: createTimer(dismissAfter, this._remove)
    }
  }

  /**
   * Run `onAdd` callback function when component is mounted
   * @returns {void}
   */
  componentDidMount() {
    const {onAdd} = this.props.notification
    if (typeof onAdd === 'function') {
      onAdd()
    }
  }

  /**
   * Run `onRemove` callback function when component will unmount
   * @returns {void}
   */
  componentWillUnmount() {
    const {onRemove} = this.props.notification
    if (typeof onRemove === 'function') {
      onRemove()
    }
  }

  /**
   * Update timer
   * @param {Object} nextProps
   * @returns {void}
   */
  componentWillReceiveProps(nextProps) {
    const {dismissAfter} = nextProps.notification
    this.setState({
      timer: createTimer(dismissAfter, this._remove)
    })
  }

  /**
   * Remove the notification
   * @private
   * @returns {void}
   */
  _remove = () => {
    const {removeNotification, notification: {id}} = this.props
    removeNotification(id)
  };

  /**
   * Pauses the timer
   * @returns {void}
   * @private
   */
  _pauseTimer = () => {
    const {timer} = this.state
    timer.pause()
  };

  /**
   * Resumes the timer
   * @returns {void}
   * @private
   */
  _resumeTimer = () => {
    const {timer} = this.state
    timer.resume()
  };

  /**
   * Wrap content in an object ready for HTML
   * @param {String} content a text
   * @returns {Object}
   * @private
   */
  _setHTML = (content) => {
    return {
      __html: content
    }
  };

  /**
   * Render button(s)
   * @returns {*}
   */
  _renderButtons = () => {
    const {
      className,
      notification: {buttons}
    } = this.props

    return buttons.map(({name, onClick, primary}) => (
      <button key={name} className={className.button} onClick={onClick}>
        <span className={className.buttonText}>
          {primary ? <b>{name}</b> : name}
        </span>
      </button>
    ))
  };

  /**
   * Render
   * @returns {XML}
   */
  render() {
    const {
      className,
      notification: {
        title,
        message,
        status,
        dismissible,
        closeButton,
        buttons,
        image,
        allowHTML
      }
    } = this.props
    const {timer} = this.state
    const notificationClass = [
      className.main,
      className.status(status),
      className.buttons(buttons.length),
      dismissible && !closeButton ? className.dismissible : null
    ].join(' ')

    if (timer) {
      this._resumeTimer()
    }

    return (
      <div
        className={className.wrapper}
        onClick={dismissible && !closeButton ? this._remove : null}
        onMouseEnter={timer ? this._pauseTimer : null}
        onMouseLeave={timer ? this._resumeTimer : null}
      >
        <div className={notificationClass}>
          {image
            ? (
              <div className={className.imageContainer}>
                <span className={className.image} style={{backgroundImage: `url(${image})`}}/>
              </div>
            ) : (
              <span className={className.icon}/>
            )
          }
          <div className={className.meta}>
            {title
              ?
              allowHTML
                ? <h4 className={className.title} dangerouslySetInnerHTML={this._setHTML(title)}/>
                : <h4 className={className.title}>{title}</h4>
              :
              null
            }
            {message
              ?
              allowHTML
                ? <p className={className.message} dangerouslySetInnerHTML={this._setHTML(message)}/>
                : <p className={className.message}>{message}</p>
              :
              null
            }
          </div>
          {dismissible && closeButton
            ? (
              <div className={className.closeButtonContainer}>
                <span className={className.closeButton} onClick={this._remove}/>
              </div>
            ) :
            null
          }
          {buttons.length
            ? (
              <div className={className.buttons()} onClick={this._remove}>
                {this._renderButtons()}
              </div>
            )
            :
            null
          }
        </div>
      </div>
    )
  }
}

export default connect(null, {removeNotification})(Notification)
