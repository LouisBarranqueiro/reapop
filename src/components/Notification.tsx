import React from 'react'

import {Notification as NotificationType} from '../reducers/notifications/types'
import {classnames} from '../constants'
import {ComponentContextType} from '../contexts/componentsContext'
import {Theme} from '..'

import {DismissNotification} from './NotificationsSystem'
import NotificationIcon from './NotificationIcon'
import NotificationDismissIcon from './NotificationDismissIcon'
import NotificationButton from './NotificationButton'
import NotificationImage from './NotificationImage'

type Props = {
    notification: NotificationType
    dismissNotification: DismissNotification
    components: ComponentContextType
    theme?: Theme
}

const Notification = (props: Props) => {
    const {notification, dismissNotification: dismiss, theme, components} = props
    const {id, title, message, dismissible, showDismissButton, buttons, allowHTML, image} = notification
    const wrapperClassname = classnames.notification(notification)
    const wrapperStyles = theme ? theme.notification(notification) : {}
    const metaStyles = theme ? theme.notificationMeta(notification) : {}
    const titleStyles = theme ? theme.notificationTitle(notification) : {}
    const messageStyles = theme ? theme.notificationMessage(notification) : {}
    const Icon = components.NotificationIcon || NotificationIcon
    const dismissNotification = () => dismiss(id)

    return (
        <div
            data-testid="notification"
            role="alert"
            style={wrapperStyles}
            className={wrapperClassname.join(' ')}
            onClick={dismissible && !showDismissButton ? dismissNotification : undefined}
        >
            <Icon theme={theme} notification={notification} />
            {image && <NotificationImage notification={notification} />}
            <div style={metaStyles} className={classnames.notificationMeta}>
                {title &&
                    (allowHTML ? (
                        <h4
                            style={titleStyles}
                            className={classnames.notificationTitle}
                            dangerouslySetInnerHTML={{__html: title}}
                        />
                    ) : (
                        <h4 style={titleStyles} className={classnames.notificationTitle}>
                            {title}
                        </h4>
                    ))}
                {message &&
                    (allowHTML ? (
                        <p
                            style={messageStyles}
                            className={classnames.notificationMessage}
                            dangerouslySetInnerHTML={{__html: message}}
                        />
                    ) : (
                        <p style={messageStyles} className={classnames.notificationMessage}>
                            {message}
                        </p>
                    ))}
            </div>
            {dismissible && showDismissButton && (
                <NotificationDismissIcon
                    notification={notification}
                    theme={theme}
                    dismissNotification={dismissNotification}
                />
            )}
            {buttons.length ? (
                <div onClick={dismissNotification} className={classnames.notificationButtons}>
                    {buttons.map((button, index) => (
                        <NotificationButton
                            key={button.name}
                            position={index}
                            button={button}
                            notification={notification}
                            theme={theme}
                        />
                    ))}
                </div>
            ) : null}
        </div>
    )
}

export default Notification
