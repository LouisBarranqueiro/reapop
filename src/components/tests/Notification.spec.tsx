import React from 'react'
import pretty from 'pretty'
import {act, render, fireEvent} from '@testing-library/react'
import {Notification as NotificationType, POSITIONS, STATUSES} from '../..'
import atalhoTheme from '../../themes/atalho'
import Notification from '../Notification'

describe('<Notification/>', () => {
    const baseNotification: NotificationType = {
        id: '1',
        title: 'Hello world!',
        message: 'say hello to the world',
        position: POSITIONS.topLeft,
        status: STATUSES.none,
        showDismissButton: true,
        buttons: [{name: 'yes'}, {name: 'no'}],
    }

    beforeEach(() => {
        jest.spyOn(atalhoTheme, 'notification').mockReturnValue({color: 'red'})
        jest.spyOn(atalhoTheme, 'notificationMeta').mockReturnValue({color: 'blue'})
        jest.spyOn(atalhoTheme, 'notificationTitle').mockReturnValue({color: 'green'})
        jest.spyOn(atalhoTheme, 'notificationMessage').mockReturnValue({color: 'white'})
    })

    afterEach(() => {
        jest.clearAllMocks()
    })

    afterAll(() => {
        jest.resetAllMocks()
    })

    it.each([
        ['without theme', undefined],
        ['with theme', atalhoTheme],
    ])('should render a notification %s', (_, theme) => {
        const {container} = render(
            <Notification
                notification={baseNotification}
                dismissNotification={jest.fn()}
                theme={theme}
                components={{}}
            />
        )
        expect(pretty(container.innerHTML)).toMatchSnapshot()
        expect((atalhoTheme.notification as jest.Mock).mock.calls).toMatchSnapshot()
        expect((atalhoTheme.notificationMeta as jest.Mock).mock.calls).toMatchSnapshot()
        expect((atalhoTheme.notificationTitle as jest.Mock).mock.calls).toMatchSnapshot()
        expect((atalhoTheme.notificationMessage as jest.Mock).mock.calls).toMatchSnapshot()
    })

    it.each([
        ['HTML not allowed', false],
        ['HTML allowed', true],
    ])('should render a notification (%s)', (_, allowHTML) => {
        const notification = {
            ...baseNotification,
            title: '<span>title</span>',
            message: '<span>message</span>',
            allowHTML,
        }
        const {container} = render(
            <Notification notification={notification} dismissNotification={jest.fn()} components={{}} />
        )
        expect(pretty(container.innerHTML)).toMatchSnapshot()
    })

    it.each([
        ['without image', undefined],
        ['with image', 'image.png'],
    ])('should render a notification %s', (_, image) => {
        const notification = {
            ...baseNotification,
            image,
        }
        const {container} = render(
            <Notification notification={notification} dismissNotification={jest.fn()} components={{}} />
        )
        expect(pretty(container.innerHTML)).toMatchSnapshot()
    })

    it.each([
        ['not dismisible', {dismissible: false}],
        ['dismissible with no dismiss button', {dismissible: true, showDismissButton: false}],
        ['dismissible with dismiss button', {dismissible: true, showDismissButton: true}],
    ])('should render a notification (%s)', (_, options) => {
        const notification = {
            ...baseNotification,
            ...options,
        }
        const {container} = render(
            <Notification notification={notification} dismissNotification={jest.fn()} components={{}} />
        )
        expect(pretty(container.innerHTML)).toMatchSnapshot()
    })

    it.each([
        ['not call', 'notification is not dismissible', {dismissible: false, showDismissButton: false}],
        ['not call', 'dismiss button is displayed', {dismissible: true, showDismissButton: true}],
        ['call', 'notification is dismissible', {dismissible: true, showDismissButton: false}],
    ])('should %s dismiss notification function on click because %s', (_, __, options) => {
        const dismissNotificationSpy = jest.fn()
        const {getByTestId} = render(
            <Notification
                notification={{...baseNotification, ...options}}
                dismissNotification={dismissNotificationSpy}
                components={{}}
            />
        )
        const notificationElem = getByTestId('notification')
        act(() => {
            fireEvent.click(notificationElem)
        })

        if (options.dismissible && !options.showDismissButton) {
            expect(dismissNotificationSpy).toHaveBeenCalledTimes(1)
        } else {
            expect(dismissNotificationSpy).not.toHaveBeenCalled()
        }
    })

    it.each([
        ['not display', 'notification is not dismissible', {dismissible: false, showDismissButton: false}],
        ['not display', 'dismiss button is not enabled', {dismissible: true, showDismissButton: false}],
        [
            'display',
            'notification is dismissible and dismiss button enabled',
            {dismissible: true, showDismissButton: true},
        ],
    ])('should %s dismiss icon because %s', (_, reason, options) => {
        const dimissNotificationSpy = jest.fn()
        const {container, getByTestId} = render(
            <Notification
                notification={{...baseNotification, ...options}}
                dismissNotification={dimissNotificationSpy}
                components={{}}
            />
        )

        expect(pretty(container.innerHTML)).toMatchSnapshot(reason)

        if (options.dismissible && options.showDismissButton) {
            act(() => {
                fireEvent.click(getByTestId('dismiss-icon'))
            })
            expect(dimissNotificationSpy).toHaveBeenCalledTimes(1)
        }
    })

    it.each([
        ['0', []],
        ['1', [{name: 'yes'}]],
        ['2', [{name: 'yes'}, {name: 'no'}]],
    ])('should render a notification with %s button(s)', (_, buttons) => {
        const {container} = render(
            <Notification
                notification={{...baseNotification, buttons}}
                dismissNotification={jest.fn()}
                components={{}}
            />
        )
        expect(pretty(container.innerHTML)).toMatchSnapshot()
    })

    it('should use the custom `NotificationIcon` component', () => {
        const CustomIcon = jest.fn().mockReturnValue(<span>custom-icon</span>)
        const {container} = render(
            <Notification
                notification={baseNotification}
                dismissNotification={jest.fn()}
                theme={atalhoTheme}
                components={{NotificationIcon: CustomIcon}}
            />
        )
        expect(container.innerHTML).toMatchSnapshot()
        expect(CustomIcon).toHaveBeenNthCalledWith(1, {notification: baseNotification, theme: atalhoTheme}, {})
    })
})
