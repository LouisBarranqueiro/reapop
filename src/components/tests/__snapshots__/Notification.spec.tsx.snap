// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification/> should display dismiss icon because notification is dismissible and dismiss button enabled: notification is dismissible and dismiss button enabled 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-2 reapop__notification--dismissible\\">
  <div class=\\"reapop__notification-meta\\">
    <h4 class=\\"reapop__notification-title\\">Hello world!</h4>
    <p class=\\"reapop__notification-message\\">say hello to the world</p>
  </div><svg class=\\"reapop__notification-dismiss-icon\\" viewBox=\\"4 4 8 8\\" fill=\\"currentColor\\" data-testid=\\"dismiss-icon\\">
    <path fill-rule=\\"evenodd\\" d=\\"M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z\\"></path>
  </svg>
  <div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">yes</span></button><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">no</span></button></div>
</div>"
`;

exports[`<Notification/> should not display dismiss icon because dismiss button is not enabled: dismiss button is not enabled 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-2 reapop__notification--dismissible\\">
  <div class=\\"reapop__notification-meta\\">
    <h4 class=\\"reapop__notification-title\\">Hello world!</h4>
    <p class=\\"reapop__notification-message\\">say hello to the world</p>
  </div>
  <div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">yes</span></button><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">no</span></button></div>
</div>"
`;

exports[`<Notification/> should not display dismiss icon because notification is not dismissible: notification is not dismissible 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-2 \\">
  <div class=\\"reapop__notification-meta\\">
    <h4 class=\\"reapop__notification-title\\">Hello world!</h4>
    <p class=\\"reapop__notification-message\\">say hello to the world</p>
  </div>
  <div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">yes</span></button><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">no</span></button></div>
</div>"
`;

exports[`<Notification/> should render a notification (HTML allowed) 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-2 \\">
  <div class=\\"reapop__notification-meta\\">
    <h4 class=\\"reapop__notification-title\\"><span>title</span></h4>
    <p class=\\"reapop__notification-message\\"><span>message</span></p>
  </div>
  <div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">yes</span></button><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">no</span></button></div>
</div>"
`;

exports[`<Notification/> should render a notification (HTML not allowed) 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-2 \\">
  <div class=\\"reapop__notification-meta\\">
    <h4 class=\\"reapop__notification-title\\">&lt;span&gt;title&lt;/span&gt;</h4>
    <p class=\\"reapop__notification-message\\">&lt;span&gt;message&lt;/span&gt;</p>
  </div>
  <div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">yes</span></button><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">no</span></button></div>
</div>"
`;

exports[`<Notification/> should render a notification (dismissible with dismiss button) 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-2 reapop__notification--dismissible\\">
  <div class=\\"reapop__notification-meta\\">
    <h4 class=\\"reapop__notification-title\\">Hello world!</h4>
    <p class=\\"reapop__notification-message\\">say hello to the world</p>
  </div><svg class=\\"reapop__notification-dismiss-icon\\" viewBox=\\"4 4 8 8\\" fill=\\"currentColor\\" data-testid=\\"dismiss-icon\\">
    <path fill-rule=\\"evenodd\\" d=\\"M4.646 4.646a.5.5 0 0 1 .708 0L8 7.293l2.646-2.647a.5.5 0 0 1 .708.708L8.707 8l2.647 2.646a.5.5 0 0 1-.708.708L8 8.707l-2.646 2.647a.5.5 0 0 1-.708-.708L7.293 8 4.646 5.354a.5.5 0 0 1 0-.708z\\"></path>
  </svg>
  <div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">yes</span></button><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">no</span></button></div>
</div>"
`;

exports[`<Notification/> should render a notification (dismissible with no dismiss button) 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-2 reapop__notification--dismissible\\">
  <div class=\\"reapop__notification-meta\\">
    <h4 class=\\"reapop__notification-title\\">Hello world!</h4>
    <p class=\\"reapop__notification-message\\">say hello to the world</p>
  </div>
  <div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">yes</span></button><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">no</span></button></div>
</div>"
`;

exports[`<Notification/> should render a notification (not dismisible) 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-2 \\">
  <div class=\\"reapop__notification-meta\\">
    <h4 class=\\"reapop__notification-title\\">Hello world!</h4>
    <p class=\\"reapop__notification-message\\">say hello to the world</p>
  </div>
  <div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">yes</span></button><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">no</span></button></div>
</div>"
`;

exports[`<Notification/> should render a notification with 0 button(s) 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-0 \\">
  <div class=\\"reapop__notification-meta\\">
    <h4 class=\\"reapop__notification-title\\">Hello world!</h4>
    <p class=\\"reapop__notification-message\\">say hello to the world</p>
  </div>
</div>"
`;

exports[`<Notification/> should render a notification with 1 button(s) 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-1 \\">
  <div class=\\"reapop__notification-meta\\">
    <h4 class=\\"reapop__notification-title\\">Hello world!</h4>
    <p class=\\"reapop__notification-message\\">say hello to the world</p>
  </div>
  <div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">yes</span></button></div>
</div>"
`;

exports[`<Notification/> should render a notification with 2 button(s) 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-2 \\">
  <div class=\\"reapop__notification-meta\\">
    <h4 class=\\"reapop__notification-title\\">Hello world!</h4>
    <p class=\\"reapop__notification-message\\">say hello to the world</p>
  </div>
  <div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">yes</span></button><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">no</span></button></div>
</div>"
`;

exports[`<Notification/> should render a notification with image 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-2 \\">
  <div class=\\"reapop__notification-image-container\\"><span class=\\"reapop__notification-image\\" style=\\"background-image: url(image.png);\\"></span></div>
  <div class=\\"reapop__notification-meta\\">
    <h4 class=\\"reapop__notification-title\\">Hello world!</h4>
    <p class=\\"reapop__notification-message\\">say hello to the world</p>
  </div>
  <div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">yes</span></button><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">no</span></button></div>
</div>"
`;

exports[`<Notification/> should render a notification with theme 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" style=\\"color: red;\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-2 \\">
  <div style=\\"color: blue;\\" class=\\"reapop__notification-meta\\">
    <h4 style=\\"color: green;\\" class=\\"reapop__notification-title\\">Hello world!</h4>
    <p style=\\"color: white;\\" class=\\"reapop__notification-message\\">say hello to the world</p>
  </div>
  <div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" style=\\"display: block; width: 100%; height: 50%; min-height: 40px; box-sizing: border-box; padding: 0px; background: rgb(47, 47, 54); border-radius: 0; border-left: 1px solid #464c5f; outline: none; text-align: center; color: rgb(255, 255, 255); cursor: pointer;\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\" style=\\"display: block; height: 25px; padding: 0px 15px; min-width: 90px; max-width: 150px; width: auto; white-space: nowrap; overflow: hidden; margin: 0px auto; text-overflow: ellipsis; text-align: center; font-size: 14px; line-height: 25px;\\">yes</span></button><button data-testid=\\"button\\" style=\\"display: block; width: 100%; height: 50%; min-height: 40px; box-sizing: border-box; padding: 0px; background: rgb(47, 47, 54); border-radius: 0; border-left: 1px solid #464c5f; outline: none; text-align: center; color: rgb(255, 255, 255); cursor: pointer; border-top: 1px solid #464c5f;\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\" style=\\"display: block; height: 25px; padding: 0px 15px; min-width: 90px; max-width: 150px; width: auto; white-space: nowrap; overflow: hidden; margin: 0px auto; text-overflow: ellipsis; text-align: center; font-size: 14px; line-height: 25px;\\">no</span></button></div>
</div>"
`;

exports[`<Notification/> should render a notification with theme 2`] = `
Array [
  Array [
    Object {
      "buttons": Array [
        Object {
          "name": "yes",
        },
        Object {
          "name": "no",
        },
      ],
      "id": "1",
      "message": "say hello to the world",
      "position": "top-left",
      "showDismissButton": true,
      "status": "none",
      "title": "Hello world!",
    },
  ],
]
`;

exports[`<Notification/> should render a notification with theme 3`] = `
Array [
  Array [
    Object {
      "buttons": Array [
        Object {
          "name": "yes",
        },
        Object {
          "name": "no",
        },
      ],
      "id": "1",
      "message": "say hello to the world",
      "position": "top-left",
      "showDismissButton": true,
      "status": "none",
      "title": "Hello world!",
    },
  ],
]
`;

exports[`<Notification/> should render a notification with theme 4`] = `
Array [
  Array [
    Object {
      "buttons": Array [
        Object {
          "name": "yes",
        },
        Object {
          "name": "no",
        },
      ],
      "id": "1",
      "message": "say hello to the world",
      "position": "top-left",
      "showDismissButton": true,
      "status": "none",
      "title": "Hello world!",
    },
  ],
]
`;

exports[`<Notification/> should render a notification with theme 5`] = `
Array [
  Array [
    Object {
      "buttons": Array [
        Object {
          "name": "yes",
        },
        Object {
          "name": "no",
        },
      ],
      "id": "1",
      "message": "say hello to the world",
      "position": "top-left",
      "showDismissButton": true,
      "status": "none",
      "title": "Hello world!",
    },
  ],
]
`;

exports[`<Notification/> should render a notification without image 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-2 \\">
  <div class=\\"reapop__notification-meta\\">
    <h4 class=\\"reapop__notification-title\\">Hello world!</h4>
    <p class=\\"reapop__notification-message\\">say hello to the world</p>
  </div>
  <div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">yes</span></button><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">no</span></button></div>
</div>"
`;

exports[`<Notification/> should render a notification without theme 1`] = `
"<div data-testid=\\"notification\\" role=\\"alert\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-2 \\">
  <div class=\\"reapop__notification-meta\\">
    <h4 class=\\"reapop__notification-title\\">Hello world!</h4>
    <p class=\\"reapop__notification-message\\">say hello to the world</p>
  </div>
  <div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">yes</span></button><button data-testid=\\"button\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\">no</span></button></div>
</div>"
`;

exports[`<Notification/> should render a notification without theme 2`] = `Array []`;

exports[`<Notification/> should render a notification without theme 3`] = `Array []`;

exports[`<Notification/> should render a notification without theme 4`] = `Array []`;

exports[`<Notification/> should render a notification without theme 5`] = `Array []`;

exports[`<Notification/> should use the custom \`NotificationIcon\` component 1`] = `"<div data-testid=\\"notification\\" role=\\"alert\\" style=\\"color: red;\\" class=\\"reapop__notification reapop__notification--none reapop__notification--buttons-2 \\"><span>custom-icon</span><div style=\\"color: blue;\\" class=\\"reapop__notification-meta\\"><h4 style=\\"color: green;\\" class=\\"reapop__notification-title\\">Hello world!</h4><p style=\\"color: white;\\" class=\\"reapop__notification-message\\">say hello to the world</p></div><div class=\\"reapop__notification-buttons\\"><button data-testid=\\"button\\" style=\\"display: block; width: 100%; height: 50%; min-height: 40px; box-sizing: border-box; padding: 0px; background: rgb(47, 47, 54); border-radius: 0; border-left: 1px solid #464c5f; outline: none; text-align: center; color: rgb(255, 255, 255); cursor: pointer;\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\" style=\\"display: block; height: 25px; padding: 0px 15px; min-width: 90px; max-width: 150px; width: auto; white-space: nowrap; overflow: hidden; margin: 0px auto; text-overflow: ellipsis; text-align: center; font-size: 14px; line-height: 25px;\\">yes</span></button><button data-testid=\\"button\\" style=\\"display: block; width: 100%; height: 50%; min-height: 40px; box-sizing: border-box; padding: 0px; background: rgb(47, 47, 54); border-radius: 0; border-left: 1px solid #464c5f; outline: none; text-align: center; color: rgb(255, 255, 255); cursor: pointer; border-top: 1px solid #464c5f;\\" class=\\"reapop__notification-button\\"><span class=\\"reapop__notification-button-text\\" style=\\"display: block; height: 25px; padding: 0px 15px; min-width: 90px; max-width: 150px; width: auto; white-space: nowrap; overflow: hidden; margin: 0px auto; text-overflow: ellipsis; text-align: center; font-size: 14px; line-height: 25px;\\">no</span></button></div></div>"`;
